### PR TITLE
Implements a crude runner

### PR DESCRIPTION
**Pull request by mikesol**
Currently only prints one error when shit hits the fan.
No way in the current architecture to elegantly pass options to the runner that percolate down to `json-schema-faker` (check out my hack in `generate.ts`...). I'd appreciate recommendations on how to make this cleaner. Once Kimmo finishes the reporter, we can hopefully link this to that.

### CODE REVIEW

File: packages/unmock-core/src/runner/index.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+const defaultRunnerOptions: IRunnerOptions = {
+  maxLoop: 20,
+};
+
+const errorHandler = (
+  allFailed: boolean,
+  errors: Error[],
+  intermediaryErrors: Array<(string | { message: string})>,
+  cb?: jest.DoneCallback) => {
```
Comment (ksaaskil) [inline packages/unmock-core/src/runner/index.ts]: TODO: This references `jest` types but the Jest types are not includes as `dependencies` of `unmock-core` so the build is not "self-contained" (also see https://github.com/unmock/unmock-js/pull/201).

File: packages/unmock-core/src/__tests__/runner.test.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      petstore = unmock.services.petstore;
+    });
+    afterAll(() => unmock.off());
+
+    beforeEach(() => {
+      petstore.reset();
+    });
+
+    test("runner loop works", runner(async () => {
+      petstore.state(withCodes(200));
```
Comment (ksaaskil) [inline packages/unmock-core/src/__tests__/runner.test.ts]: This is not even using the runner, or?

File: packages/unmock-core/src/runner/index.ts (hunk 1 above)
Comment (mikesol) [reply packages/unmock-core/src/runner/index.ts]: Yeah, we probably want to do some tests with mocha and also look into peer dependencies.

File: packages/unmock-core/src/__tests__/runner.test.ts (hunk 2 above)
Comment (mikesol) [reply packages/unmock-core/src/__tests__/runner.test.ts]: `test("runner loop works", runner(async () => { `

**Pull request merged**